File: 1604-least-number-of-unique-integers-after-k-removals/least-number-of-unique-integers-after-k-removals.py
from collections import Counter
class Solution:
    def findLeastNumOfUniqueInts(self, arr: List[int], k: int) -> int:
        c=Counter(arr)
        y=OrderedDict(c.most_common())
        m=[]
        p=[]
        for i in y.keys():
            m.append(i)
            p.append(y[i]) 
        l=len(m)
        for i in range(len(p)-1,-1,-1):
            if(p[i]<=k):
                k=k-p[i]
                l=l-1
            else:
                return l
        return 0

File: 1604-least-number-of-unique-integers-after-k-removals/least-number-of-unique-integers-after-k-removals.cpp
class Solution {
public:
    vector<int>x;
    vector<int>p;
    void sort(map<int,int>&mpp)
    {
        multimap<int,int>m;
        for(auto& it:mpp)
        {
            m.insert({it.second,it.first});
        }
        for(auto i:m)
        {
            x.push_back(i.first);
            p.push_back(i.second);
        }
    }
    int findLeastNumOfUniqueInts(vector<int>& arr, int k) {
        map<int,int>mpp;
        for(int i=0;i<arr.size();i++)
        {
            mpp[arr[i]]++;
        }
        sort(mpp);
        int len=p.size();
        for(int i=0;i<x.size();i++)
        {
            if(x[i]<=k)
            {
                k=k-x[i];
                len=len-1;
            }
            else{
                return len;
            }
        }
        return 0;
    }
};